src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpTagHelperRunnerInitializationVisitor.cs

```diff
@@ -26,6 +26,25 @@ public class CSharpTagHelperRunnerInitializationVisitor : CodeVisitor<CSharpCode
             _tagHelperContext = Context.Host.GeneratedClassContext.GeneratedTagHelperContext;
         }
 
+        /// <inheritdoc />
+        public override void Accept(Chunk chunk)
```

Comment:
> Since you're updating the @using here as well, this problem impacts other directives as well, for instance: any @model or @addTagHelper inside of sections, statements etc. won't be picked up.

Previously `@model`, `@addTagHelper`, and `@using` inside statements or sections were _not_ picked up. Now they are.

Diving into a few extra blocks isn't a big deal -- see existing `CSharpTagHelperFieldDeclarationVisitor` for example.